### ISSUE TITLE
apm-server only listens on ipv4 when using 'localhost:8200' as host

### ISSUE THREAD

**Issue by prupert** [elastic/apm-server#1405]
I am using the apm-server-6.4.1-1.x86_64 package for CentOS 7 and noticed that the `apm-server` process only binds to the IPv4 address when using `localhost`, even though 'localhost' resolves to both IPv4 and IPv6, and IPv6 is fully enabled and working. 

config:
```
apm-server:
  host: "localhost:8200"
```

```
# host localhost
localhost has address 127.0.0.1
localhost has IPv6 address ::1

# netstat -tulpn | grep apm
tcp        0      0 127.0.0.1:8200          0.0.0.0:*               LISTEN      12345/apm-server
```

For other Elastic services, i.e. Elasticsearch, the process does bind to IPv6 as well:
```
# grep network.host /etc/elasticsearch/elasticsearch.yml
network.host: _local_
# netstat -tulpn | grep 9200
tcp6       0      0 127.0.0.1:9200          :::*                    LISTEN      23456/java
tcp6       0      0 ::1:9200                :::*                    LISTEN      23456/java
```

**Comment by graphaelli** [comment]
apm-server currently listens on only one of the resolved addresses when the listener is specified by host name, preferring ipv4, per the [underlying library](https://golang.org/pkg/net/#Listen) implementation.

We can turn this into a feature request supporting multiple listeners, perhaps `apm-server.hosts` as an array, but this does not qualify as a bug.

Some options for working around this:

* listen only with ipv6 - `apm-server.host="[::1]:8200"`
* listen on all interfaces -  `apm-server.host=:8200` - be sure to filter traffic per interface per your requirements

**Comment by prupert** [comment]
Thanks for the clarification. In that case might I suggest to use `127.0.0.1` by default in stead of `localhost`, because the latter is very confusing. See also the documentation of the underlying library implementation you mentioned:
>  "The address can use a host name, **but this is not recommended**, because it will create a listener for at most one of the host's IP addresses."

Users of APM Server should be made aware that they should not use the hostname `localhost` to connect to  the APM Server when IPv6 is enabled.

**Comment by OrangeDog** [comment]
I've tried both `0.0.0.0:8200` and `:8200`, but the server only listens on IPv6 and not IPv4.
I'd say that's definitely a bug.

Only by explicitly putting an assigned IPv4 address will it listen correctly.

**Comment by graphaelli** [comment]
@OrangeDog Are you verifying through netstat or sending ipv4 traffic?  I don't expect an issue with the latter, since https://codereview.appspot.com/2204048 originally addressed the issue.

If you are unable to complete an v4 request to a dual stack listening on `:8200`, please provide your OS and apm-server version + installation method (package or source) so that we may attempt to reproduce the issue.

**Comment by OrangeDog** [comment]
Checking with netstat. Sorry, you're right it's dual stack and does work fine.

**Comment by cwiechmann** [comment]
I'm also running into this issue using the official Docker-Image (Helm-Chart) to deploy the APM-Server. Version 7.15.0.

I have configured the following:
```
      apm-server:
        host: "0.0.0.0:8200"
```

In the logs I can see the following, which looks like IPv6 and I'm not able to make an IPv4 request:
`Listening on: [::]:8200","service.name":"apm-server","ecs.version":"1.6.0"`

FYI: 
I tried with an older version 7.4 and that was additionally reporting the following:
`host resolved from 0.0.0.0:8200 to [::]:8200`

On another system, not having ipv6 enabled all looks good:
`Listening on: 0.0.0.0:8200","service.name":"apm-server","ecs.version":"1.6.0"`

**Comment by stuartnelson3** [comment]
Closing this as users can bind to all interfaces to listen on ipv4 and ipv6
